cmp/grid/impl/MenuSupport.ts

```diff
@@ -270,3 +259,53 @@ function replaceHoistToken(token: string, gridModel: GridModel): Some<RecordActi
             return token;
     }
 }
+
+function createExpandCollapseItem(gridModel: GridModel): RecordAction[] {
+    if (!gridModel || gridModel.maxDepth === 0) return null;
+
+    return [
+        new RecordAction({
+            text: 'Expand All',
+            icon: Icon.groupRowExpanded(),
+            actionFn: () => gridModel.expandAll()
+        }),
+        new RecordAction({
+            text: 'Collapse All',
+            icon: Icon.groupRowCollapsed(),
+            actionFn: () => gridModel.collapseAll()
+        }),
+        levelExpandAction(gridModel)
+    ];
+}
+
+function levelExpandAction(gridModel: GridModel): RecordAction {
+    return new RecordAction({
+        text: 'Expand to ...',
+        displayFn: () => {
+            // Don't show for degenerate shallow grid models, or if we don't have labels
+            const {maxDepth, expandToLevel} = gridModel;
+            if (maxDepth <= 1) return {hidden: true};
+
+            const levelLabels = executeIfFunction(gridModel.levelLabels);
+            if (!levelLabels || levelLabels.length < maxDepth + 1) {
+                gridModel.logDebug(
+                    '"levelLabels" not provided or of insufficient length. No menu items shown.'
```

Comment:
we're sure we don't want to give Level 1, Level 2, ect generic labels by default?

Reply:
Spoke to Lee.